src/VncClient.cpp

```diff
@@ -364,9 +382,15 @@ void VncClient::processClientData()
                 m_data->state = Rfb::Init;
             else
             {
-                auto r = QRandomGenerator::system();
                 m_data->challenge.resize( 16 );
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
+                auto r = QRandomGenerator::system();
                 r->generate( m_data->challenge.begin(), m_data->challenge.end() );
+#else
+                typedef std::independent_bits_engine<std::default_random_engine, CHAR_BIT, uchar> random_bytes_engine;
+                random_bytes_engine randomEngine;
+                std::generate( m_data->challenge.begin(), m_data->challenge.end(), std::ref(randomEngine) );
```

Comment:
Isn't std::generate a C++-17 feature, while Qt 5.6 only requires C++-11 ?
Maybe it is better to use qrand() instead ?

Reply:
Yes, you are right.